lib/src/http/managers/webhook_manager.dart

```diff
@@ -0,0 +1,293 @@
+    final messageManager = MessageManager(
+      client.options.messageCacheConfig,
+      client,
+      channelId: Snowflake.parse((response.jsonBody as Map<String, Object?>)['channel_id'] as String),
+    );
+    final message = messageManager.parse(response.jsonBody as Map<String, Object?>);
+
+    messageManager.cache[message.id] = message;
+    return message;
+  }
+
+  /// Update a message sent by a webhook.
+  Future<Message> updateWebhookMessage(
+    Snowflake webhookId,
+    Snowflake messageId,
+    MessageUpdateBuilder builder, {
+    required String token,
+    Snowflake? threadId,
+  }) async {
+    final route = HttpRoute()
+      ..webhooks(id: webhookId.toString())
+      ..add(HttpRoutePart(token))
+      ..messages(id: messageId.toString());
+
+    final queryParameters = {if (threadId != null) 'thread_id': threadId.toString()};
+    final HttpRequest request;
+    if (!identical(builder.attachments, sentinelList) && builder.attachments?.isNotEmpty == true) {
+      final attachments = builder.attachments!;
+      final payload = builder.build();
+
+      final files = <MultipartFile>[];
+      for (int i = 0; i < attachments.length; i++) {
+        files.add(MultipartFile.fromBytes(
+          'files[$i]',
+          attachments[i].data,
+          filename: attachments[i].fileName,
+        ));
+
+        ((payload['attachments'] as List)[i] as Map)['id'] = i.toString();
+      }
+
+      request = MultipartRequest(
+        route,
+        method: 'PATCH',
+        jsonPayload: jsonEncode(payload),
+        files: files,
+        queryParameters: queryParameters,
+        authenticated: false,
+      );
+    } else {
+      request = BasicRequest(
+        route,
+        method: 'PATCH',
+        body: jsonEncode(builder.build()),
+        queryParameters: queryParameters,
+        authenticated: false,
+      );
+    }
+
+    final response = await client.httpHandler.executeSafe(request);
+    final messageManager = MessageManager(
+      client.options.messageCacheConfig,
+      client,
+      channelId: Snowflake.parse((response.jsonBody as Map<String, Object?>)['channel_id'] as String),
+    );
+    final message = messageManager.parse(response.jsonBody as Map<String, Object?>);
+
+    messageManager.cache[message.id] = message;
+    return message;
+  }
+
+  /// Delete a message sent by a webhook.
+  Future<void> deleteWebhookMessage(Snowflake webhookId, Snowflake messageId, {required String token, Snowflake? threadId}) async {
+    final route = HttpRoute()
+      ..webhooks(id: webhookId.toString())
+      ..add(HttpRoutePart(token))
+      ..messages(id: messageId.toString());
+    final request = BasicRequest(
+      route,
+      method: 'DELETE',
+      queryParameters: {if (threadId != null) 'thread_id': threadId.toString()},
+      authenticated: false,
+    );
+
+    await client.httpHandler.executeSafe(request);
+    // TODO: Can we delete the message from the cache?
```

Comment:
I think message should be deleted here, yes

Reply:
Yeah - the issue is we don't have the channel id to get the correct cache to remove it from

Reply:
Then let's leave it for later. Doesn't matter that much I think

Reply:
We'll probably also get a gatetway event for the message deletion, so we can handle it there.